lib/net.c

```diff
@@ -48,31 +49,218 @@ find_num_numa_nodes(void)
 }
 
 static void
-close_num_ports(uint8_t nb_ports)
+close_iface_ports(struct gatekeeper_if *iface, uint8_t nb_ports)
 {
-	uint8_t port_id;
+	uint8_t i;
 
-	for (port_id = 0; port_id < nb_ports; port_id++) {
-		rte_eth_dev_stop(port_id);
-		rte_eth_dev_close(port_id);
+	for (i = 0; i < nb_ports; i++) {
+		rte_eth_dev_stop(iface->ports[i]);
+		rte_eth_dev_close(iface->ports[i]);
 	}
 }
 
+static void
+close_iface_id(struct gatekeeper_if *iface, uint8_t nb_slave_ports)
+{
+	uint8_t i;
+
+	/* If there's only one port, there's no bonded port. */
+	if (iface->num_ports == 1)
+		return;
+
+	for (i = 0; i < nb_slave_ports; i++)
+		rte_eth_bond_slave_remove(iface->id, iface->ports[i]);
+
+	rte_eth_bond_free(iface->name);
+}
+
+static void
+close_iface(struct gatekeeper_if *iface)
+{
+	close_iface_id(iface, iface->num_ports);
+	close_iface_ports(iface, iface->num_ports);
```

Comment:
During initialization, if initializing some of the ports fails, `close_iface_id` and `close_iface_ports` are called with a second parameter that designates how many successful port openings there were. So I intended to re-use that code and call it with all the ports here.